### ISSUE TITLE
Examples do not work

### ISSUE THREAD

**Issue by anselal** [toxicrecker/DiscordUtils#22]
Hi there, I installed the latest version of discordUtils==1.28, discord.py==1.7.0 and python 3.8
I tried the examples and they do not print anything. also I do not get any errors in my console.

**Comment by anselal** [comment]
More specific I tried the `paginate` commands

**Comment by toxicrecker** [comment]
It works for me

**Comment by anselal** [comment]
Yeah, well, this ain't helping me. You build the library. Maybe some help here ?
Are there any special permissions my bot has to have ? I gave read/write and read history permissions, and reaction permisions

**Comment by MiyakoYakota** [comment]
> Yeah, well, this ain't helping me. You build the library. Maybe some help here ?
> Are there any special permissions my bot has to have ? I gave read/write and read history permissions, and reaction permisions

Just because you use someone's code does not mean you are entitled to support. Post your code, and pray some person with time on their hands will be willing to help you.

**Comment by anselal** [comment]
Well, the cide us taken from the examples as I already mentioned. Besides that, you 2 had the time being rude but haven't the time to help? Thus is a open source project isn't it? Shame on both of you @MiyakoYakota @toxicrecker

**Comment by MiyakoYakota** [comment]
> Well, the code is taken from the examples as I already mentioned. Besides that, you 2 had the time being rude but haven't the time to help? This is an open source project isn't it? Shame on both of you @MiyakoYakota @toxicrecker 

Again, post your code, and hope someone will help you.

**Comment by anselal** [comment]
Again, my code is the Project's examples. Never mind

**Comment by MiyakoYakota** [comment]
> Again, my code is the Project's examples. Never mind

The samples work fine. You are doing something wrong. Post your code, last time I will attempt.

**Comment by anselal** [comment]
@MiyakoYakota 

```python
@bot.command()
async def paginate(ctx):
    embed1 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 1")
    embed2 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 2")
    embed3 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 3")
    paginator = DiscordUtils.Pagination.CustomEmbedPaginator(ctx)
    paginator.add_reaction('⏮️', "first")
    paginator.add_reaction('⏪', "back")
    paginator.add_reaction('🔐', "lock")
    paginator.add_reaction('⏩', "next")
    paginator.add_reaction('⏭️', "last")
    embeds = [embed1, embed2, embed3]
    await paginator.run(embeds)
```

**Comment by remagnedlogeht** [comment]
> @MiyakoYakota
> 
> ```python
> @bot.command()
> async def paginate(ctx):
>     embed1 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 1")
>     embed2 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 2")
>     embed3 = discord.Embed(color=ctx.author.color).add_field(name="Example", value="Page 3")
>     paginator = DiscordUtils.Pagination.CustomEmbedPaginator(ctx)
>     paginator.add_reaction('⏮️', "first")
>     paginator.add_reaction('⏪', "back")
>     paginator.add_reaction('🔐', "lock")
>     paginator.add_reaction('⏩', "next")
>     paginator.add_reaction('⏭️', "last")
>     embeds = [embed1, embed2, embed3]
>     await paginator.run(embeds)
> ```

Can you show us what error you had while using that code?

**Comment by anselal** [comment]
no error on the console. but the command did not return anything. maybe it has to do with bot permissions ? I already asked this but didn't get a serious response

**Comment by remagnedlogeht** [comment]
> no error on the console. but the command did not return anything. maybe it has to do with bot permissions ? I already asked this but didn't get a serious response

Then what permissions did your bot have? since it would need embed links permission to send the embed

**Comment by toxicrecker** [comment]
> no error on the console. but the command did not return anything. maybe it has to do with bot permissions ? I already asked this but didn't get a serious response

Do you have an error handler that raises the errors for you?
in case you dont:
```py
@bot.event
async def on_command_error(ctx, error):
    raise error
```

**Comment by anand2312** [comment]
> > no error on the console. but the command did not return anything. maybe it has to do with bot permissions ? I already asked this but didn't get a serious response
> 
> Do you have an error handler that raises the errors for you?
> in case you dont:
> 
> ```python
> @bot.event
> async def on_command_error(ctx, error):
>     raise error
> ```

This is a rather useless error handler. If the user hasn't overridden the default provided `on_command_error`, [and there was an error, it'd be printed to the console](https://github.com/Rapptz/discord.py/blob/9f3551926ad5176c0cbf23a1a127452a2749a135/discord/ext/commands/bot.py#L157)

The only way that errors would be getting suppressed would be if they had _faulty_ error handler; one which handled only a few cases and silently ignored the rest, like:
```py
@bot.event
async def on_command_error(ctx, error):
    if isinstance(error, ...):
        # does something
```
And the correct way would be....
```py
@bot.event
async def on_command_error(ctx, error):
    if isinstance(error, ...):
        # does something
    raise error  # this makes sure that you re-raise any errors that you weren't planning on handling
```

**Comment by JayDwee** [comment]
Did you install all requirements for discord.py[voice] 

```bash
$ sudo apt install libffi-dev python-dev
$ pip3 install PyNaCl
$ pip3 install DiscordUtils
```

I have also put in a PR to remove this requirement #25 in which this issue (if it is what you're having) will no longer need those additional requirements

**Comment by anselal** [comment]
I did not install pynacl nor libffi-dev

**Comment by JayDwee** [comment]
> I did not install pynacl nor libffi-dev

That'll be the issue 😄 

You can either install those dependencies, or if you are unable to you can use my branch while waiting for #25 to be approved with
```bash
$ pip install git+https://github.com/KoalaBotUK/DiscordUtils
```

glad to help

**Comment by anselal** [comment]
There is no reason to install a voice library to make a pagination work. I still think it is a permissions issue.

What permissions are required to make it work ?

**Comment by JayDwee** [comment]
It's to do with the original import, when importing DiscordUtils it imports the Music class which will crash if discord[voice] is not successfully downloaded.

Please try the suggested fix and let us know if it solves your problem so this issue can be closed.

**Comment by JayDwee** [comment]
@anselal DiscordUtils has been updated now so try installing with `pip install DiscordUtils` and let us know if your code starts working so this issue can be closed
